### PR TITLE
Release note for simplex tree options

### PR DESCRIPTION
**Pull request by VincentRouvreau**

### CODE REVIEW

File: .github/next_release.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 We are pleased to announce the release 3.9.0 of the GUDHI library.
 
-As a major new feature, the GUDHI library now offers ...
+As a major new feature, the GUDHI library now offers a much faster implementation for the Cubical 2d case, fast cofaces access and stable simplex handles for the `Simplex_tree`.
```
[the remark this answers is not in the document]
Comment (VincentRouvreau) [reply .github/next_release.md:3]: This is a kind of *punchline* I am also using for the web site
image

File: .github/next_release.md [hunk 2]
```diff
@@ -9,8 +9,23 @@ Below is a list of changes made since GUDHI 3.8.0:
...
+     - From GUDHI 3.9.0, `Simplex_tree_options_full_featured` now activates `link_nodes_by_label` and `stable_simplex_handles`.
+
+     | Simplex_tree_options_*  | :warning: full_featured | default | python | minimal |
+     | ---- | ---- | ---- | ---- | ---- |
+     | store_key              | 1       | 1      | 1      | 0 |
+     | store_filtration       | 1       | 1      | 1      | 0 |
+     | contiguous_vertices    | 0       | 0      | 0      | 0 |
+     | link_nodes_by_label    | ***1*** | 0      | 0      | 0 |
+     | stable_simplex_handles | ***1*** | 0      | 0      | 0 |
+     | Filtration_value       | double  | double | double | 0 |
```
[the remark this answers is not in the document]
Comment (VincentRouvreau) [reply .github/next_release.md]: Removes 0 as `Filtration_value` type on [ed022e7](https://github.com/GUDHI/gudhi-devel/pull/1005/commits/ed022e7e9e49c608e257254a4550022ffa653010)

File: .github/next_release.md [hunk 3]
```diff
@@ -9,8 +9,23 @@ Below is a list of changes made since GUDHI 3.8.0:
 - [Module](link)
      - **...**
 
-- [Module](link)
-     - **...**
+- [Simplex_tree options](https://gudhi.inria.fr/doc/latest/struct_simplex_tree_options.html)
+     - A new option `link_nodes_by_label` to enhance cofaces and stars access, when set to true.
+     - A new option `stable_simplex_handles` to keep Simplex handles even after insertions or removals, when set to true.
```
[the remark this answers is not in the document]
Comment (VincentRouvreau) [reply .github/next_release.md]: Add valid on [ed022e7](https://github.com/GUDHI/gudhi-devel/pull/1005/commits/ed022e7e9e49c608e257254a4550022ffa653010)

File: .github/next_release.md [hunk 4]
```diff
@@ -9,8 +9,23 @@ Below is a list of changes made since GUDHI 3.8.0:
 - [Module](link)
      - **...**
 
-- [Module](link)
-     - **...**
+- [Simplex_tree options](https://gudhi.inria.fr/doc/latest/struct_simplex_tree_options.html)
+     - A new option `link_nodes_by_label` to enhance cofaces and stars access, when set to true.
```
[the remark this answers is not in the document]
Comment (VincentRouvreau) [reply .github/next_release.md]: Replace enhance by speed up on [ed022e7](https://github.com/GUDHI/gudhi-devel/pull/1005/commits/ed022e7e9e49c608e257254a4550022ffa653010)

**Comment by mglisse** [comment]
Not (necessarily) in this PR, but I think we should also mention #919 #912 #909 #903 #896 #997 in the changes.

File: .github/next_release.md [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
+- [Simplex_tree options](https://gudhi.inria.fr/doc/latest/struct_simplex_tree_options.html)
+     - A new option `link_nodes_by_label` to speed up cofaces and stars access, when set to true.
+     - A new option `stable_simplex_handles` to keep Simplex handles valid even after insertions or removals, when set to true.
 
-- [Module](link)
-     - **...**
+- [Simplex_tree](https://gudhi.inria.fr/doc/latest/group__simplex__tree.html)
+     - `Simplex_tree_options_full_featured` has been renamed `Simplex_tree_options_default` and `Simplex_tree_options_python`.
+     These are respectively the options used by all the utilities and by the python interface of the `SimplexTree` (as before this version).
+     - From GUDHI 3.9.0, `Simplex_tree_options_full_featured` now activates `link_nodes_by_label` and `stable_simplex_handles`.
```
Comment (mglisse) [inline .github/next_release.md:16]: > The default parameter is now called `Simplex_tree_options_default` (same behavior as before), while `Simplex_tree_options_full_featured` now enables both `link_nodes_by_label` and `stable_simplex_handles`, making it slower. It remains recommended to use a custom option class that defines exactly the options you need (and derives from `Simplex_tree_options_minimal` for compatibility with future releases).

? (it isn't true that all utilities use "default", some use "fast")

File: .github/next_release.md [hunk 6]
```diff
@@ -1,23 +1,47 @@
...
+
+- [Simplex_tree options](https://gudhi.inria.fr/doc/latest/struct_simplex_tree_options.html)
+     - A new option `link_nodes_by_label` to speed up cofaces and stars access, when set to true.
+     - A new option `stable_simplex_handles` to keep Simplex handles valid even after insertions or removals, when set to true.
+
+- [Čech complex](https://gudhi.inria.fr/doc/latest/group__cech__complex.html)
+     - A function `assign_MEB_filtration` that assigns to each simplex a filtration value equal to the squared radius of its minimal enclosing ball (MEB), given a simplicial complex and an embedding of its vertices. Applied on a Delaunay triangulation, it computes the Delaunay-Čech filtration.
+
+- [Edge collapse](https://gudhi.inria.fr/python/latest/edge_collapse.html)
+     - A Python function `collapse_edges` to simplify a clique filtration (represented as a sparse weighted graph), while preserving its persistent homology.
```
[the remark this answers is not in the document]
Comment (VincentRouvreau) [reply .github/next_release.md]: Was renamed `reduce_graph`. Renamed on [7b0f255](https://github.com/GUDHI/gudhi-devel/pull/1005/commits/7b0f255fe4f3ddf0aa04a7b18b873d73ac4ebf18)

File: .github/next_release.md [hunk 7]
```diff
@@ -1,23 +1,47 @@
...
-
-- [Module](link)
-     - **...**
-
 - [CubicalPersistence](https://gudhi.inria.fr/python/latest/cubical_complex_sklearn_itf_ref.html)
      - Much faster implementation for the 2d case with input from top-dimensional cells.
 
+- [Simplex_tree](https://gudhi.inria.fr/doc/latest/group__simplex__tree.html)
+     - A helper `for_each_simplex` that applies a given function object on each simplex
+     - `make_filtration_non_decreasing` has been rewritten, and a new method `num_simplices_by_dimension` is now available thanks to this helper.
```
[the remark this answers is not in the document]
Comment (VincentRouvreau) [reply .github/next_release.md]: I removed `make_filtration_non_decreasing` on [7b0f255](https://github.com/GUDHI/gudhi-devel/pull/1005/commits/7b0f255fe4f3ddf0aa04a7b18b873d73ac4ebf18)

File: .github/next_release.md (hunk 5 above)
Comment (VincentRouvreau) [reply .github/next_release.md]: I precised when `Simplex_tree_options_default` was used by the `Simplex_tree` on [7b0f255](https://github.com/GUDHI/gudhi-devel/pull/1005/commits/7b0f255fe4f3ddf0aa04a7b18b873d73ac4ebf18)

Comment (mglisse) [reply .github/next_release.md]: I wonder if we shouldn't include something like the speed warning of my paragraph above :thinking:

Comment (VincentRouvreau) [reply .github/next_release.md]: @mglisse I added a speed warning on [d872578](https://github.com/GUDHI/gudhi-devel/pull/1005/commits/d872578d1768ac8d0257787ffe96e5636ed50a17)

**Review by mglisse — APPROVED** [review]

**Pull request merged**